nntrainer/tensor/char_tensor.cpp

```diff
@@ -268,6 +267,56 @@ void CharTensor::initialize(Initializer init) {
   initialize();
 }
 
+int CharTensor::multiply_i(float const &value) {
+  // multiply value to scale factors
+  float *g_scale = (float *)getScale();
+
+  sscal(scale_size(), value, g_scale, 1);
+  return ML_ERROR_NONE;
+}
+
+Tensor &CharTensor::multiply(Tensor const &input, Tensor &output,
+                             const float scale) const {
+  NNTR_THROW_IF(input.getFormat() != this->getFormat(), std::invalid_argument)
+    << "Tensor Format of " << getName() << ":"
+    << ((bool)(this->getFormat()) ? "NHWC" : "NCHW") << " is not match. ("
+    << ((bool)(input.getFormat()) ? "NHWC" : "NCHW") << ")";
+
+  NNTR_THROW_IF(!contiguous || !input.getContiguous() ||
+                  !output.getContiguous(),
+                std::invalid_argument)
+    << getName() << " is not contiguous, cannot multiply";
+
+  float lhs_scale = *(float *)getScale();
+  float rhs_scale = *input.getScale<float>();
```

Comment:
Are they always scalars? Based on #2844, it seems scale can be an array.
Is it related to the condition in the note, i.e.,  *4. only per-tensor quantization qscheme is supported* ?
if so, we need to add a condition to verify if the case can be supported or not.

Reply:
Thank you for the detailed review! Yes, you're correct. Currently, there isn't such a way to check the quantization scheme of the tensor. I'll create a new PR to do so and apply it.